src/App.js

```diff
@@ -1,41 +1,68 @@
 import React, { useEffect } from 'react';
-import {Router, Route, Switch, Redirect, Link, withRouter} from 'react-router-dom';
+import {Router, Route, Switch, Redirect} from 'react-router-dom';
 import { createBrowserHistory } from 'history';
 import { connect } from 'react-redux';
 
 import { getUserData } from './actions';
-import LoginPage from './pages/LoginPage';
 import Layout from './components/Layout';
-import {HomePage, NewQuestionPage, QuestionDetailPage, QuestionsPage} from "./pages";
+import {NotFoundPage} from "./pages";
+import routes from "./routes";
 
 export const history = createBrowserHistory();
 
-function AuthorizedApp() {
+function App({ user, getUserData }) {
+  useEffect(() => { getUserData(); }, []);
+
+  return user.id === undefined ? (<p>Loading...</p>) : (
+    <Router history={history}>
+      <Switch>
+
+        {routes.map((route, index) => {
+          let TargetComponent = route.component;
+
+          let RouteComponent = Route;
+          if (route.loginRequired)
+            RouteComponent = LoggedInRoute;
+          else if (route.logoutRequired)
+            RouteComponent = LoggedOutRoute;
+
+          return (
+            <RouteComponent user={user} path={route.path} exact key={index}>
+              <LayoutWrapper route={route}>
+                <TargetComponent />
+              </LayoutWrapper>
+            </RouteComponent>
+          );
+        })}
+
+        <Route path='*'><NotFoundPage/></Route>
+      </Switch>
+    </Router>
```

Comment:
Szerintem egyszerűbb lenne, ha filterrel végigmennél a logged In meg a logout route-okon.